cms/djangoapps/contentstore/views/course.py

```diff
@@ -1646,3 +1650,24 @@ def _get_course_creator_status(user):
         course_creator_status = 'granted'
 
     return course_creator_status
+
+
+def _get_library_creator_status(user):
+    """
+    Helper method for returning the library creation status for a particular user,
+    taking into account the values of DISABLE_LIBRARY_CREATION and LIBRARIES_ENABLED.
+
+    """
+
+    if not LIBRARIES_ENABLED:
+        library_creator_status = 'disallowed_for_this_site'
+    elif user.is_staff:
+        library_creator_status = 'granted'
+    elif CourseCreatorRole().has_user(user):
+        library_creator_status = 'granted'
+    elif settings.FEATURES.get('DISABLE_LIBRARY_CREATION', False):
```

Comment:
What's the use case for this new flag, `DISABLE_LIBRARY_CREATION`? It's confusingly named: it sounds like it completely disables creation of content libraries for all users, but we can see here that global staff or course creators can still create libraries regardless of this setting. (I know you're copying `DISABLE_COURSE_CREATION` but that's also confusingly named and poorly documented).

As implemented, the setting also has no effect whatsoever: whether `DISABLE_LIBRARY_CREATION` is True or False, this method will still return `'disallowed_for_this_site'`.

I think you should just remove `DISABLE_LIBRARY_CREATION` completely, and use something like this:

```
    if not LIBRARIES_ENABLED:
        return False
    elif user.is_staff:
        return True
    elif settings.FEATURES.get('ENABLE_CREATOR_GROUP', False):
        return CourseCreatorRole().has_user(user)
    else:
        return True
```